daemon/hostdaemon.go

```diff
@@ -31,25 +32,33 @@ type HostDaemon struct {
 
 func (d *HostDaemon) CreateBridgePort(pf int, vf int, vlan int, mac string) (error) {
 	d.ensureConnected()
+
+	macBytes, err := hex.DecodeString(mac)
+	if err != nil {
+		return err
+	}
+
 	createRequest := &pb.CreateBridgePortRequest{
 		BridgePort: &pb.BridgePort{
-			Name: fmt.Sprintf("%d-%d-%d-%s", pf, vf, vlan, mac),
+			Name: fmt.Sprintf("%d-%d", pf, vf),
 			Spec: &pb.BridgePortSpec{
 				Ptype:          1,
-				MacAddress:     []byte{},
-				LogicalBridges: []string{},
+				MacAddress:     macBytes,
+			    LogicalBridges: []string{
+					fmt.Sprintf("%d", vlan),
+				},
 			},
 		},
 	}
 
-	_, err := d.client.CreateBridgePort(context.TODO(), createRequest)
+	_, err = d.client.CreateBridgePort(context.TODO(), createRequest)
 	return err
 }
 
 func (d *HostDaemon) DeleteBridgePort(pf int, vf int, vlan int, mac string) (error) {
 	d.ensureConnected()
 	req := &pb.DeleteBridgePortRequest{
-		    Name: fmt.Sprintf("%d-%d-%d-%s", pf, vf, vlan, mac),
+		Name: fmt.Sprintf("%d-%d-%d-%s", pf, vf, vlan, mac),
```

Comment:
The call to DeleteBridgePort isn't the same. I'm marking it as resolved since we don't need to change this right now but we might want to revisit this.